Errors when building wheel using pip

ERROR: Command errored out with exit status 1:
     command: 'C:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\evange76\\AppData\\Local\\Temp\\pip-install-_8q3wlxh\\isosplit5\\setup.py'"'"'; __file__='"'"'C:\\Users\\evange76\\AppData\\Local\\Temp\\pip-install-_8q3wlxh\\isosplit5\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\evange76\AppData\Local\Temp\pip-record-hj3uwexu\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\Include\isosplit5'
         cwd: C:\Users\evange76\AppData\Local\Temp\pip-install-_8q3wlxh\isosplit5\
    Complete output (25 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.7
    creating build\lib.win-amd64-3.7\isosplit5
    copying isosplit5\__init__.py -> build\lib.win-amd64-3.7\isosplit5
    running build_ext
    building 'isosplit5_interface' extension
    creating build\temp.win-amd64-3.7
    creating build\temp.win-amd64-3.7\Release
    creating build\temp.win-amd64-3.7\Release\src
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IC:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\Library\include -IC:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\Library\include -IC:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\include -IC:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /EHsc /Tpsrc/isosplit5.cpp /Fobuild\temp.win-amd64-3.7\Release\src/isosplit5.obj /EHsc /DVERSION_INFO=\"0.1.3\"
    isosplit5.cpp
    C:\Users\evange76\AppData\Local\Temp\pip-install-_8q3wlxh\isosplit5\src\isocut5.h(21): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
    C:\Users\evange76\AppData\Local\Temp\pip-install-_8q3wlxh\isosplit5\src\isocut5.h(21): error C2146: syntax error: missing ';' before identifier 'bigint'
    C:\Users\evange76\AppData\Local\Temp\pip-install-_8q3wlxh\isosplit5\src\isocut5.h(27): error C2061: syntax error: identifier 'bigint'
    C:\Users\evange76\AppData\Local\Temp\pip-install-_8q3wlxh\isosplit5\src\isosplit5.h(30): error C2061: syntax error: identifier 'bigint'
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\stdint.h(17): error C2371: 'int64_t': redefinition; different basic types
    C:\Users\evange76\AppData\Local\Temp\pip-install-_8q3wlxh\isosplit5\src\isocut5.h(21): note: see declaration of 'int64_t'
    src/isosplit5.cpp(23): error C2923: 'std::vector': 'bigint' is not a valid template type argument for parameter '_Ty'
    C:\Users\evange76\AppData\Local\Temp\pip-install-_8q3wlxh\isosplit5\src\isocut5.h(21): note: see declaration of 'bigint'
    src/isosplit5.cpp(23): fatal error C1903: unable to recover from previous error(s); stopping compilation
    Internal Compiler Error in C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\bin\HostX86\x64\cl.exe.  You will be prompted to send an error report to Microsoft later.
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.24.28314\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\evange76\\AppData\\Local\\Temp\\pip-install-_8q3wlxh\\isosplit5\\setup.py'"'"'; __file__='"'"'C:\\Users\\evange76\\AppData\\Local\\Temp\\pip-install-_8q3wlxh\\isosplit5\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\evange76\AppData\Local\Temp\pip-record-hj3uwexu\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\evange76\AppData\Local\Continuum\anaconda3\envs\spike\Include\isosplit5' Check the logs for full command output.

--- a/.gitignore
+++ b/.gitignore
@@ -8,3 +8,5 @@ dist/
 *.egg-info/
 
 .idea
+.eggs
+tmp/

--- a/.gitignore
+++ b/.gitignore
@@ -6,3 +6,5 @@ var/
 build/
 dist/
 *.egg-info/
+
+.idea

--- a/setup.py
+++ b/setup.py
@@ -100,5 +100,6 @@ setup(
     cmdclass={'build_ext': BuildExt},
     install_requires=[
         'numpy','pybind11>=2.2'
-    ]
+    ],
+    setup_requires=['pybind11>=2.2']
 )

--- a/src/isocut5.h
+++ b/src/isocut5.h
@@ -17,6 +17,7 @@
 #define ISOCUT5_H
 
 #include <stdlib.h>
+#include <cstdint>
 
 typedef int64_t bigint;
 

--- a/src/isosplit5.cpp
+++ b/src/isosplit5.cpp
@@ -313,9 +313,7 @@ bool isosplit5(int* labels, bigint M, bigint N, float* X, isosplit5_opts opts)
     ns_isosplit5::compute_covmats(covmats, M, N, Kmax, X, labels, centroids, clusters_to_compute_vec);
 
     // The active labels are those that are still being used -- for now, everything is active
-    int active_labels_vec[Kmax];
-    for (bigint i = 0; i < Kmax; i++)
-        active_labels_vec[i] = 1;
+    std::vector<int> active_labels_vec(Kmax, 1);
     std::vector<int> active_labels;
     for (bigint i = 0; i < Kmax; i++)
         active_labels.push_back(i + 1);
@@ -435,9 +433,7 @@ bool isosplit5(int* labels, bigint M, bigint N, float* X, isosplit5_opts opts)
     }
 
     // We should remap the labels to occupy the first natural numbers
-    bigint labels_map[Kmax];
-    for (bigint i = 0; i < Kmax; i++)
-        labels_map[i] = 0;
+    std::vector<bigint> labels_map(Kmax, 0);
     for (bigint i = 0; i < (bigint)active_labels.size(); i++) {
         labels_map[active_labels[i] - 1] = i + 1;
     }
@@ -896,8 +892,9 @@ void get_pairs_to_compare(std::vector<bigint>* inds1, std::vector<bigint>* inds2
 {
     inds1->clear();
     inds2->clear();
-    double dists[K][K];
+    std::vector<std::vector<double>> dists(K);
     for (bigint k1 = 0; k1 < K; k1++) {
+        dists[k1].resize(K);
         for (bigint k2 = 0; k2 < K; k2++) {
             if ((active_comparisons_made[k1][k2]) || (k1 == k2))
                 dists[k1][k2] = -1;
@@ -1187,11 +1184,10 @@ void compare_pairs(std::vector<bigint>* clusters_changed, bigint* total_num_labe
 
 void get_pairs_to_compare3(std::vector<bigint>* i1s, std::vector<bigint>* i2s, bigint M, bigint N, double* centroids)
 {
-    float distances[N][N];
-    bigint used[N];
-    for (bigint i = 0; i < N; i++)
-        used[i] = 0;
+    std::vector<bigint> used(N, 0);   
+    std::vector<std::vector<float>> distances(N);
     for (bigint i = 0; i < N; i++) {
+        distances[i].resize(N);
         for (bigint j = i; j < N; j++) {
             double sumsqr = 0;
             for (bigint m = 0; m < M; m++) {
@@ -1205,9 +1201,7 @@ void get_pairs_to_compare3(std::vector<bigint>* i1s, std::vector<bigint>* i2s, b
     bool something_changed = true;
     while (something_changed) {
         something_changed = false;
-        bigint closest[N];
-        for (bigint i = 0; i < N; i++)
-            closest[i] = -1;
+        std::vector<bigint> closest(N, -1);
         for (bigint i = 0; i < N; i++) {
             double best_distance = -1;
             if (!used[i]) {

--- a/src/jisotonic5.h
+++ b/src/jisotonic5.h
@@ -18,6 +18,7 @@
 #define jisotonic5_h
 
 #include <stdlib.h>
+#include <cstdint>
 
 typedef int64_t bigint;
 

